src/Sylius/Bundle/CoreBundle/Resources/config/services/promotion.xml

```diff
@@ -17,6 +17,12 @@
             <argument type="service" id="sylius.repository.promotion" />
         </service>
 
+        <service id="sylius.promotion_processor" class="Sylius\Component\Promotion\Processor\PromotionProcessor">
```

Comment:
Sounds like a plan! That way we ensure that proper provider is injected everywhere else 👍